i6_models/config.py

```diff
@@ -0,0 +1,51 @@
+from dataclasses import MISSING, dataclass
+from typing import Any, List, Optional
+
+
+@dataclass
+class ModelConfiguration:
+    """base dataclass that supported fetching attributes"""
+
+    _name: Optional[str] = None
+
+    @staticmethod
+    def name() -> Optional[str]:
+        return None
+
+    def _get_all_attributes(self) -> List[str]:
+        return [k for k in self.__dataclass_fields__.keys()]
+
+    def _get_name(self, attribute_name: str) -> str:
+        return self.__dataclass_fields__[attribute_name].name
+
+    def _get_default(self, attribute_name: str) -> Any:
+        if hasattr(self, attribute_name):
+            return getattr(self, attribute_name)
+
+        f = self.__dataclass_fields__[attribute_name]
+        if f.default_factory is not MISSING:
+            return f.default_factory()
+        return f.default
+
+    def _get_type(self, attribute_name: str) -> Any:
+        return self.__dataclass_fields__[attribute_name].type
+
+    @classmethod
+    def from_namespace(cls, args):
```

Comment:
Can you please add type annotation?

Reply:
I am not 100% sure about the typing myself. So from what I understand this can be any object on which has attributes, so maybe `Any` is correct here?

Reply:
Yes, `Any` should be OK.